urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/yuccasrv/commandline/YuccaCommandFactory.java

```diff
@@ -0,0 +1,133 @@
+package ikrs.yuccasrv.commandline;
+
+/**
+ * @author Ikaros Kappler
+ * @date 2012-05-07
+ * @version 1.0.0
+ **/
+
+import java.text.ParseException;
+
+
+import ikrs.typesystem.BasicType;
+import ikrs.util.DefaultCommandFactory;
+import ikrs.util.Command;
+import ikrs.util.CommandFactory;
+import ikrs.util.CommandStringIncompleteException;
+import ikrs.util.DefaultCommand;
+import ikrs.util.UnknownCommandException;
+import ikrs.yuccasrv.Yucca;
+
+public class YuccaCommandFactory
+    extends DefaultCommandFactory
+    implements CommandFactory<Command> {
+
+
+    /* The actual yucca server */
+    private Yucca
+	server;
+
+    /**
+     * The constructor.
+     *
+     * The server instanced will be delegated to all commands that are made with this factory.
+     *
+     * @param server If available pass the server instance; if not available, just pass null.
+     *
+     **/
+    public YuccaCommandFactory( Yucca server ) {
+	super();
+
+	this.server = server;
+
+	
+	this.addSupportedCommand( new CommandHelp( this, null, null ) );   // The help command needs to know its factory
+	this.addSupportedCommand( new CommandLicense() );
+	this.addSupportedCommand( new CommandListen() );
+	this.addSupportedCommand( new CommandLogLevel() );
+	this.addSupportedCommand( new CommandQuit() );
+	this.addSupportedCommand( new CommandStatus() );
+	this.addSupportedCommand( new CommandUnlisten() );
+	this.addSupportedCommand( new CommandVersion() );
+	this.addSupportedCommand( new CommandWarranty() );
+    }
+
+    /**
+     * Get the factories server instance.
+     *
+     * Warning: it is not guaranteed that the server is set; in this case the method returns null.
+     *
+     * @return The YuccaServer instance or null if not set.
+     **/
+    protected Yucca getServer() {
+	return this.server;
+    }
+
+
+    //---BEGIN----------------------- AbstractCommandFactory implementation -----------------------
+    /**
+     * Make a new Command with the given name and params.
+     *
+     * @param name The command's name.
+     * @param params The command's params - in BasicType representation.
+     * 
+     * @return The new command.
+     **/
+    public Command make( String name,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.